File: AntlrGrammarEditor/AntlrRuntimes/CSharp/AntlrCaseInsensitiveInputStream.cs
using Antlr4.Runtime;

class AntlrCaseInsensitiveInputStream : AntlrInputStream
{
    private string lookaheadData;

    public AntlrCaseInsensitiveInputStream(string input, bool lowerCase = false)
        : base(input)
    {
        lookaheadData = lowerCase ? input.ToLowerInvariant() : input.ToUpperInvariant();
    }

    public override int LA(int i)
    {
        if (i == 0)
        {
            return 0; // undefined
        }
        if (i < 0)
        {
            i++; // e.g., translate LA(-1) to use offset i=0; then data[p+0-1]
            if (p + i - 1 < 0)
            {
                return IntStreamConstants.EOF; // invalid; no char before first char
            }
        }

        if (p + i - 1 >= n)
        {
            return Eof;
        }

        return lookaheadData[p + i - 1];
    }
}

File: AntlrGrammarEditor/AntlrRuntimes/Go/AntlrCaseInsensitiveInputStream.go
package main

import (
    "github.com/antlr/antlr4/runtime/Go/antlr"
    "strings"
)

type AntlrCaseInsensitiveInputStream struct {
    *antlr.InputStream
    lookaheadData []rune
}

func NewAntlrCaseInsensitiveInputStream(input string) *AntlrCaseInsensitiveInputStream {
    caseInsenstiveStream := new(AntlrCaseInsensitiveInputStream)
    caseInsenstiveStream.InputStream = antlr.NewInputStream(input)
    caseInsenstiveStream.lookaheadData = []rune(strings.ToLower(input))
    return caseInsenstiveStream
}

func (is *AntlrCaseInsensitiveInputStream) LA(offset int) int {
    if offset == 0 {
        return 0 // nil
    }
    if offset < 0 {
        offset++ // e.g., translate LA(-1) to use offset=0
    }
    var pos = is.Index() + offset - 1
    
    if pos < 0 || pos >= is.Size() { // invalid
        return antlr.TokenEOF
    }
    
    return int(is.lookaheadData[pos])
}